lib/cdo/script_constants.rb

```diff
@@ -165,6 +165,37 @@ module ScriptConstants
 
   DEFAULT_VERSION_YEAR = '2017'
 
+  # A whitelist of all family names for scripts.
+  FAMILY_NAMES = [
+    # CSF
+    COURSEA = 'coursea'.freeze,
+    COURSEB = 'courseb'.freeze,
+    COURSEC = 'coursec'.freeze,
+    COURSED = 'coursed'.freeze,
+    COURSEE = 'coursee'.freeze,
+    COURSEF = 'coursef'.freeze,
+    EXPRESS = 'express'.freeze,
+    PREEXPRESS = 'pre-express'.freeze,
+
+    # CSP
+    CSP1 = 'csp1'.freeze,
+    CSP2 = 'csp2'.freeze,
+    CSP3 = 'csp3'.freeze,
+    CSP4 = 'csp4'.freeze,
+    CSP5 = 'csp5'.freeze,
```

Comment:
nothing to do here right now... but this reminds me that the idea of "csp5" script family makes less sense over time, considering CSP 20-21 may have a totally different structure. ideally... we wouldn't even need these script families, and would rely on the course families, which control both (1) where you get redirected to from /courses/csd and (2) what shows up in the assignment selector. Just food for thought.